internal/queue/types_test.go

```diff
@@ -0,0 +1,249 @@
+			},
+			expectErr:       chErr,
+			expectChCalls:   1,
+			expectConnCalls: 1,
+		},
+		{
+			name: "connection error only",
+			client: &Client{
+				amqpChannel:    &fakeChannel{isClosedRet: false},
+				amqpConnection: &fakeConnection{isClosedRet: false, closeErr: connErr},
+			},
+			expectErr:       connErr,
+			expectChCalls:   1,
+			expectConnCalls: 1,
+			expectChClosed:  true,
+			expectConnClose: true,
+		},
+		{
+			name:            "nil resources",
+			client:          &Client{},
+			expectErr:       nil,
+			expectChCalls:   0,
+			expectConnCalls: 0,
+		},
+		{
+			name: "already closed resources",
+			client: &Client{
+				amqpChannel:    &fakeChannel{isClosedRet: true},
+				amqpConnection: &fakeConnection{isClosedRet: true},
+			},
+			expectErr:       nil,
+			expectChCalls:   0,
+			expectConnCalls: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.client.Close()
+			if tt.expectErr != nil {
+				assert.ErrorIs(t, err, tt.expectErr)
+			} else {
+				assert.NoError(t, err)
+			}
+
+			// Inspect internal fakes when present
+			if fc, ok := tt.client.amqpChannel.(*fakeChannel); ok {
+				assert.Equal(t, tt.expectChCalls, fc.closeCalls)
+				if tt.expectChClosed {
+					assert.True(t, fc.closed)
+				}
+			}
+			if fconn, ok := tt.client.amqpConnection.(*fakeConnection); ok {
+				assert.Equal(t, tt.expectConnCalls, fconn.closeCalls)
+				if tt.expectConnClose {
+					assert.True(t, fconn.closed)
+				}
+			}
+		})
+	}
+}
+
+func TestAmqpConsumer_Table(t *testing.T) {
+	t.Run("Close delegates to client", func(t *testing.T) {
+		t.Parallel()
+		ch := &fakeChannel{isClosedRet: false}
+		conn := &fakeConnection{isClosedRet: false}
+
+		cons := &AmqpConsumer{
+			queueName: "q",
+			client: &Client{
+				amqpChannel:    ch,
+				amqpConnection: conn,
+			},
+			db:     &fakeJobDB{},
+			logger: slog.Default(),
+		}
+
+		err := cons.Close()
+		assert.NoError(t, err)
+		assert.Equal(t, 1, ch.closeCalls)
+		assert.Equal(t, 1, conn.closeCalls)
+	})
+
+	t.Run("Start exits on context cancellation", func(t *testing.T) {
+		t.Parallel()
+		msgs := make(chan amqp.Delivery)
+		ch := &fakeChannel{isClosedRet: false, consumeCh: msgs}
+		conn := &fakeConnection{isClosedRet: false}
+
+		cons := &AmqpConsumer{
+			queueName: "test-queue",
+			client: &Client{
+				amqpChannel:    ch,
+				amqpConnection: conn,
+			},
+			db:     &fakeJobDB{},
+			logger: slog.Default(),
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		errCh := make(chan error, 1)
+
+		go func() { errCh <- cons.Start(ctx) }()
+		cancel()
+
+		select {
+		case err := <-errCh:
+			assert.ErrorIs(t, err, context.Canceled)
+		case <-time.After(2 * time.Second):
+			t.Fatal("consumer did not exit on context cancellation")
+		}
+	})
+}
+
+func TestClient_Close_Idempotent(t *testing.T) {
+	ch := &fakeChannel{isClosedRet: false}
```

Comment:
Can you use the arrange/act/assert comment as mentioned in https://github.com/canonical/github-runner-operators/blob/main/STYLE.md#comments-in-tests ?